Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -0,0 +1,544 @@
+
+        if constexpr (TIFF_DEBUG) {
+            if (tiff_value.size() == 1) {
+                tiff_value[0].visit(
+                    [&](auto const& value) {
+                        dbgln("Read tag({}), type({}): {}", tag, to_underlying(type), value);
+                    });
+            } else {
+                dbg("Read tag({}), type({}): [", tag, to_underlying(type));
+                for (u32 i = 0; i < tiff_value.size(); ++i) {
+                    tiff_value[i].visit(
+                        [&](auto const& value) {
+                            dbg("{}", value);
+                        });
+                    if (i != tiff_value.size() - 1)
+                        dbg(", ");
+                }
+                dbgln("]");
+            }
+        }
+
+        TRY(handle_tag(tag, type, count, tiff_value));
+
+        return {};
+    }
+
+    ErrorOr<void> handle_tag(u16 tag, Type type, u32 count, Vector<Value> const& value)
+    {
+        // FIXME: Make that easy to extend
+        switch (tag) {
+        case 256:
+            // ImageWidth
+            if ((type != Type::Short && type != Type::Long) || count != 1)
+                return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid tag 256");
+
+            value[0].visit(
+                [this]<OneOf<u16, u32> T>(T const& width) {
+                    m_size.set_width(width);
+                },
+                [&](auto const&) {
+                    VERIFY_NOT_REACHED();
+                });
+            break;
+
+        case 257:
+            // ImageLength
+            if ((type != Type::Short && type != Type::Long) || count != 1)
+                return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid tag 257");
+
+            value[0].visit(
+                [this]<OneOf<u16, u32> T>(T const& width) {
+                    m_size.set_height(width);
+                },
+                [&](auto const&) {
+                    VERIFY_NOT_REACHED();
+                });
+            break;
+
+        case 258:
+            // BitsPerSample
+            if (type != Type::Short || count != 3)
+                return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid tag 258");
+
+            for (u8 i = 0; i < m_bits_per_sample.size(); ++i) {
+                value[i].visit(
+                    [this, i](u16 const& bits_per_sample) {
+                        m_bits_per_sample[i] = bits_per_sample;
+                    },
+                    [&](auto const&) {
+                        VERIFY_NOT_REACHED();
+                    });
+            }
+            break;
+
+        case 259:
+            // Compression
+            if (type != Type::Short || count != 1)
+                return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid tag 259");
+
+            TRY(value[0].visit(
+                [this](u16 const& compression) -> ErrorOr<void> {
+                    switch (compression) {
+                    case 1:
```

Comment:
I'd do

```
switch (static_cast<Compression>(compression)) {
  case Compression::NoCompression:
  case Compression::CCITT:
  ....
    m_compression = static_cast<Compression>(compression);
  default:
    return Error::...
```

That way, you don't need to repeat the enum values here.

Reply:
I think that this is even simpler:
```cpp
if (compression > 6 && compression != to_underlying(Compression::PackBits))
	return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid compression value");

m_compression = static_cast<Compression>(compression);
return {};
```